packages/examples/cvat/recording-oracle/src/handlers/process_intermediate_results.py

```diff
@@ -980,17 +389,121 @@ def process_intermediate_results(  # noqa: PLR0912
             ", ".join(f"{k}: {v:.2f}" for k, v in job_results.items()),
         )
 
-    if validation_result.updated_gt_stats:
-        updated_gt_stats = _compute_gt_stats_update(
-            initial_gt_stats, validation_result.updated_gt_stats
-        )
+    gt_stats = validation_result.gt_stats
+    if gt_stats:
+        cvat_task_id_to_failed_val_frames: dict[
+            int, set[int]
+        ] = {}  # cvat_task_id: {val_frame_id, ...}
+        rejected_job_ids = rejected_jobs.keys()
+
+        if rejected_job_ids:
+            job_id_to_task_id = {j.job_id: j.task_id for j in unchecked_jobs_meta.jobs}
+            job_id_to_frame_range = {j.job_id: j.job_frame_range for j in unchecked_jobs_meta.jobs}
+
+            # find validation frames to be disabled
+            for rejected_job_id in rejected_job_ids:
+                job_frame_range = job_id_to_frame_range[rejected_job_id]
+                cvat_task_id = job_id_to_task_id[rejected_job_id]
+                honeypots_mapping = validation_result.task_id_to_honeypots_mapping[cvat_task_id]
+                job_honeypots = set(honeypots_mapping.keys()) & set(job_frame_range)
+                validation_frames = [
+                    val_frame
+                    for honeypot, val_frame in honeypots_mapping.items()
+                    if honeypot in job_honeypots
+                ]
+
+                for val_frame in validation_frames:
+                    val_frame_stats = gt_stats[(cvat_task_id, val_frame)]
+                    if (
+                        val_frame_stats.failed_attempts >= Config.validation.gt_ban_threshold
+                        and not val_frame_stats.accepted_attempts
+                    ):
+                        cvat_task_id_to_failed_val_frames.setdefault(cvat_task_id, set()).add(
+                            val_frame
+                        )
```

Comment:
If I understand correctly, we can have the same GT frame in different tasks, and they can have different statistics, so that the same frame can be banned in 1 task, but not another. It doesn't look right, can we change gt stats to be escrow-wise?

Reply:
done